File: TransCoder_tokenized_test_set_functions/python/SMALLEST_SUBARRAY_K_DISTINCT_NUMBERS.py
def minRange ( arr, n, k ) :
    l = 0
    r = n
    for i in range ( n ) :
        s = [ ]
        for j in range ( i, n ) :
            s . append ( arr [ j ] )
            if ( len ( s ) == k ) :
                if ( ( j - i ) < ( r - l ) ) :
                    r = j
                    l = i
                break
        if ( j == n ) :
            break
    if ( l == 0 and r == n ) :
        print ( "Invalid k" )
    else :
        print ( l, r )
    

File: TransCoder_tokenized_test_set_functions/cpp/SMALLEST_SUBARRAY_K_DISTINCT_NUMBERS.cpp
void minRange ( int arr [ ], int n, int k ) {
  int l = 0, r = n;
  for ( int i = 0;
  i < n;
  i ++ ) {
    unordered_set < int > s;
    int j;
    for ( j = i;
    j < n;
    j ++ ) {
      s . insert ( arr [ j ] );
      if ( s . size ( ) == k ) {
        if ( ( j - i ) < ( r - l ) ) {
          r = j;
          l = i;
        }
        break;
      }
    }
    if ( j == n ) break;
  }
  if ( l == 0 && r == n ) cout << "Invalid k";
  else cout << l << " " << r;
}